src/kvstore/raftex/Host.cpp

```diff
@@ -300,6 +291,47 @@ void Host::appendLogsInternal(folly::EventBase* eb,
                 }
                 return r;
             }
+            case cpp2::ErrorCode::E_WAITING_SNAPSHOT: {
+                VLOG(2) << self->idStr_
+                        << "The host is waiting for the snapshot, so we need to send log from "
+                        << " current committedLogId " << self->committedLogId_;
+                std::shared_ptr<cpp2::AppendLogRequest> newReq;
+                cpp2::AppendLogResponse r;
+                {
+                    std::lock_guard<std::mutex> g(self->lock_);
+                    auto res = self->checkStatus();
+                    if (res != cpp2::ErrorCode::SUCCEEDED) {
+                        VLOG(2) << self->idStr_
+                                << "The host is not in a proper status,"
+                                   " skip waiting the snapshot";
+                        r.set_error_code(res);
+                        self->setResponse(r);
+                    } else {
+                        self->lastLogIdSent_ = self->committedLogId_;
+                        self->lastLogTermSent_ = self->logTermToSend_;
+                        newReq = self->prepareAppendLogRequest();
+                        r = std::move(resp);
+                    }
+                }
+                if (newReq) {
+                    self->appendLogsInternal(eb, newReq);
+                } else {
+                    self->noMoreRequestCV_.notify_all();
+                }
+                return r;
+            }
+            case cpp2::ErrorCode::E_LOG_STALE: {
+                VLOG(2) << self->idStr_ << "Log stale, reset lastLogIdSent " << self->lastLogIdSent_
+                        << " to the followers lastLodId" << resp.get_last_log_id();
+                {
+                    std::lock_guard<std::mutex> g(self->lock_);
+                    self->lastLogIdSent_ = resp.get_last_log_id();
+                    self->lastLogTermSent_ = resp.get_last_log_term();
+                    self->setResponse(resp);
+                }
+                self->noMoreRequestCV_.notify_all();
+                return resp;
```

Comment:
maybe the lambda no need return a response object.

Reply:
Yes,  you are right.  Let me remove all unnecessary response in the callback.  
Good point.